src/helpers/google.ts

```diff
@@ -0,0 +1,164 @@
+    // Get spreadsheet metadata (to get all sheet names)
+    const spreadsheetInfo = await sheets.spreadsheets.get({ spreadsheetId: documentId });
+    if (!spreadsheetInfo.data.sheets) {
+      console.log("No sheets to fetch!");
+      return;
+    }
+    const sheetNames = spreadsheetInfo.data.sheets.map((sheet) => sheet.properties?.title).filter((el) => !!el);
+
+    let text = "";
+
+    // Fetch data from each sheet
+    for (const sheetName of sheetNames) {
+      const result = await sheets.spreadsheets.values.get({
+        spreadsheetId: documentId,
+        range: `${sheetName}`,
+      });
+
+      const rows = result.data.values;
+      if (!rows || rows.length === 0) {
+        console.log(`No data found in sheet: ${sheetName}`);
+        continue;
+      }
+
+      text += String(rows.map((row) => String(row.flat()) + "\n"));
+    }
+    return {
+      title: spreadsheetInfo.data.properties?.title,
+      data: text,
+    };
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const getContentFromDocs = async (documentId: string) => {
+  const auth = await authenticate();
+  const docs = google.docs({ version: "v1", auth });
+
+  try {
+    const res = await docs.documents.get({ documentId: documentId });
+    const doc = res.data;
+    if (!doc.body || !doc.body.content) {
+      throw new Error("Error fetching from docs");
+    }
+
+    let text = "";
+    // Extract text from the document body
+    doc.body.content.forEach((element) => {
+      if (element.paragraph && element.paragraph.elements) {
+        element.paragraph.elements.forEach((paraElement) => {
+          if (paraElement.textRun) {
+            text += paraElement.textRun.content;
+          }
+        });
+      }
+    });
+
+    return {
+      title: doc.title,
+      data: text,
+    };
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const getContentFromSlides = async (documentId: string) => {
+  const auth = await authenticate();
+
+  const slides = google.slides({ version: "v1", auth });
+
+  const res = await slides.presentations.get({ presentationId: documentId });
+  const presentation = res.data;
+
+  if (!presentation.slides) {
+    console.log("Nothign to fetch");
+    return;
+  }
+
+  let text = "";
+
+  presentation.slides.forEach((slide) => {
+    if (!slide.pageElements) return;
+    slide.pageElements.forEach((element) => {
+      if (element.shape && element.shape.text && element.shape.text.textElements) {
+        element.shape.text.textElements.forEach((textElement) => {
+          if (textElement.textRun && textElement.textRun.content) {
+            text += textElement.textRun.content;
+          }
+        });
+      }
+    });
+  });
+
+  return {
+    title: presentation.title,
+    data: text,
+  };
+};
+
+export const getContentFromUrl = async (url: string) => {
+  const { docType, documentId } = consumeUrl(url);
+  switch (docType) {
+    case "SHEETS":
+      return await getContentFromSpreadsheet(documentId);
```

Comment:
```suggestion
      return await getContentFromSheets(documentId);
```
Seems more standardized naming